### PR TITLE
chore: add bad-credentials exception for install script

### LINKED ISSUE #5755: Handling Authentication Errors in install_kustomize.sh`
### What happened?

The install_kustomize.sh script is designed to download and install the kustomize binary. However, if a `GITHUB_TOKEN` is set in the environment but is invalid or expired, the script fails with authentication errors, such as Bad credentials. 

```
{
  "message": "Bad credentials",
  "documentation_url": "https://docs.github.com/rest",
  "status": "401"
}
```

Currently, the script attempts to authenticate with GitHub using the `GITHUB_TOKEN` but does not handle cases where this token is incorrect or invalid, leading to confusing error messages like:
```shell
Version v5.4.3 does not exist or is not available for linux/amd64.
```

If the support for the token is expected in the installation script, extending the validation to consider such errors sounds reasonable. I assume this extra block would be enough, at least to give the end user some guidance (mostly when inspecting CI logs):

```bash
# Check for authentication errors
if [[ $releases == *"Bad credentials"* ]]; then
  echo "Authentication failed: Invalid GITHUB_TOKEN. Please check or remove your token."
  exit 1
```
https://github.com/kubernetes-sigs/kustomize/blob/27d508fa039d9f5f03c6ae666114e4c138690c2f/hack/install_kustomize.sh#L144-L147

### What did you expect to happen?

The script should handle authentication errors gracefully and provide a clear message indicating that there is an issue with the `GITHUB_TOKEN`.

### How can we reproduce it (as minimally and precisely as possible)?

1. Set an invalid or expired `GITHUB_TOKEN` in your environment.
2. Run the [install_kustomize.sh](https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh) script locally passing a desired version: 
   - `bash install_kustomize.sh 5.4.3`
4. Observe the script's failure due to authentication errors from GitHub.

### Expected output

downloaded binary

### Actual output

```shell
Version v5.4.3 does not exist or is not available for linux/amd64.
```

### Kustomize version

5.4.3

### Operating system

Linux

### PR DESCRIPTION
**Pull request by viniciusdc**
closes #5755 

Improved error handling for cases where the GITHUB_TOKEN is invalid or expired. The `install_kustomize.sh` script now provides a clear message when authentication fails due to token issues.

### CODE REVIEW

**Comment by stormqueen1990** [comment]
Hi there, @viniciusdc!

Could you please sign the CNCF CLA? We cannot accept contributions unless you sign it.

Instructions on how to do so can be found here: https://kubernetes.io/docs/contribute/new-content/#sign-the-cla

Can you also please update your commit messages to follow the Conventional Commit standard, as described [here](https://github.com/kubernetes-sigs/kustomize/blob/27d508fa039d9f5f03c6ae666114e4c138690c2f/CONTRIBUTING.md#pull-request-rules)?

Thanks in advance!

**Comment by viniciusdc** [comment]
Hi @stormqueen1990, thanks for the quick followback. I am just waiting for the approval for the CLA from my company. Thanks for pointing out the docs for the commit standard; I went to look for something like that before opening the PR but didn't find it; I will amend it later today.

File: hack/install_kustomize.sh
Comment (chansuke) [inline hack/install_kustomize.sh:147]:
```suggestion
if [[ $releases == *"Bad credentials"* ]]; then
  echo "Authentication failed: Invalid GITHUB_TOKEN. Please check or remove your token."
  exit 1
fi  
```

I think you need to close the `if` condition

Comment (viniciusdc) [reply hack/install_kustomize.sh]: Thanks @chansuke!

**Review by stormqueen1990 — COMMENTED** [review]
Thanks @viniciusdc!

/lgtm
/assign @varshaprasad96
for maintainer review

**Review by chansuke — APPROVED** [review]

**Comment by viniciusdc** [comment]
Thanks, @chansuke @stormqueen1990, for the initial review! Pinging @varshaprasad96 for final approval.

**Comment by viniciusdc** [comment]
Hi, @chansuke @stormqueen1990, is there anything left on my side regarding commands to the bot that I might need to include?

**Comment by stormqueen1990** [comment]
> Hi, @chansuke @stormqueen1990, is there anything left on my side regarding commands to the bot that I might need to include?

Hi there, @viniciusdc! The next step would be the maintainer review.

**Review by varshaprasad96 — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
